bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/ApiHandler.java

```diff
@@ -105,7 +105,7 @@ public void dispose() {
         }
     }
 
-    public synchronized <T> T executeUri(URI uri, HttpMethod method, Class<T> clazz, @Nullable String sessionToken,
+    public <T> T executeUri(URI uri, HttpMethod method, Class<T> clazz, @Nullable String sessionToken,
```

Comment:
I believe the idea here was to avoid concurrent requests to the Freebox API. It was probably done for a certain reason. @clinique can you confirm ?
This is not really clear for me if this is still the case with your changes. Do you do it through other lock mechanism ?

Reply:
I just ask myself why the method was synchronized.

Reply:
I just created https://github.com/openhab/openhab-core/issues/5033 as I really miss these annotations, and think they could be very helpful to others as well.